mutmut/__init__.py

```diff
@@ -238,31 +362,38 @@ def decorator_mutation(children, **_):
     return children[-1:]
 
 
-def trailer_mutation(children, **_):
-    if len(children) == 3 and children[0].type == 'operator' and children[0].value == '[' and children[-1].type == 'operator' and children[-1].value == ']' and children[0].parent.type == 'trailer' and children[1].type == 'name' and children[1].value != 'None':
-        # Something that looks like "foo[bar]"
-        return [children[0], Name(value='None', start_pos=children[0].start_pos), children[-1]]
-    return children
+array_subscript_pattern = ASTPattern("""
+_name[_any]
+#       ↑
+""")
 
 
-def arglist_mutation(children, **_):
-    if len(children) > 3 and children[0].type == 'name' and children[0].value != 'None':
-        return [Name(value='None', start_pos=children[0].start_pos)] + children[1:]
-    return children
+function_call_pattern = ASTPattern("""
+_name(_any)
```

Comment:
What happens if we write?
```python
function_call_pattern = ASTPattern("""
_name(_any)
#        ↑
```

Maybe to avoid confusion it should be required to cover `_any` fully like so:
```python
function_call_pattern = ASTPattern("""
_name(_any)
#     ↑↑↑↑
```

Reply:
Pointing at any place on the variable should work fine. 

I have some ideas that we could use this tooling interactively and then it would be awkward to fully cover things you want to match.